projects/packages/search/src/dashboard/components/pages/dashboard-page.jsx

```diff
@@ -68,10 +68,19 @@ export default function DashboardPage( { isLoading = false } ) {
 
 	// Introduce the gate for new pricing with URL parameter `new_pricing_202208=1`
 	const isNewPricing = useSelect( select => select( STORE_ID ).isNewPricing202208(), [] );
-	const isJustUpgraded = useSelect(
-		select => select( STORE_ID ).isFeatureEnabled( 'just_upgraded' ),
-		[]
-	);
+
+	const myStorage = window.localStorage;
+	const isTooltipRead = myStorage.getItem( 'upgrade_tooltip_finished' );
+	const setTooltipRead = () => {
+		myStorage.setItem( 'upgrade_tooltip_finished', 1 );
+	};
+	const isPlanJustUpgraded =
+		useSelect( select => select( STORE_ID ).isPlanJustUpgraded(), [] ) && ! isTooltipRead;
+
+	const tooltipHelper = {
+		isPlanJustUpgraded,
+		setTooltipRead,
+	};
```

Comment:
Let's avoid this pattern for propagating a small number of properties. `tooltipHelper` is significantly less informative than propagating `setTooltipRead` and `isPlanJustUpgraded` directly.

Reply:
Yes. Just tried to make nested props passing more concise. 🤔